test/server/cards/02_SHD/leaders/TheMandalorianSwornToTheCreed.spec.ts

```diff
@@ -0,0 +1,125 @@
+describe('The Mandalorian, Sworn To The Creed', function () {
+    integration(function (contextRef) {
+        describe('The Mandalorian\'s leader ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['green-squadron-awing', 'waylay'],
+                        leader: 'the-mandalorian#sworn-to-the-creed',
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'battlefield-marine']
+                    },
+                });
+            });
+
+            it('should not exhaust units when not playing upgrades', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.greenSquadronAwing);
+                expect(context.player2).toBeActivePlayer();
+
+                context.player2.passAction();
+
+                context.player1.clickCard(context.waylay);
+                context.player1.clickCard(context.wampa);
+                expect(context.player2).toBeActivePlayer();
+            });
+        });
+
+        describe('The Mandalorian\'s leader ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['academy-training'],
+                        groundArena: ['wampa'],
+                        resources: ['armed-to-the-teeth', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst', 'atst'],
+                        leader: 'the-mandalorian#sworn-to-the-creed',
+                    },
+                    player2: {
+                        hand: ['vambrace-flamethrower'],
+                        groundArena: ['academy-defense-walker', 'battlefield-marine'],
```

Comment:
Can we add a unit that has high hp and some damage so that it still has remainingHp <= 4. Same comment for the deployed test.